indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -393,17 +392,6 @@ public Response getTaskStatus(@PathParam("taskid") String taskid)
     return Response.status(status).entity(response).build();
   }
 
-  @Deprecated
-  @GET
-  @Path("/task/{taskid}/segments")
-  @Produces(MediaType.APPLICATION_JSON)
-  @ResourceFilters(TaskResourceFilter.class)
-  public Response getTaskSegments(@PathParam("taskid") String taskid)
-  {
-    final Set<DataSegment> segments = taskStorageQueryAdapter.getInsertedSegments(taskid);
-    return Response.ok().entity(segments).build();
```

Comment:
i wonder if we should leave this endpoint with a helpful error message about this functionality being removed

Reply:
Sure, that makes sense. It would be better than just throwing a 404.

Reply:
Updated.